bundles/org.eclipse.equinox.launcher.tests/src/org/eclipse/equinox/launcher/tests/LauncherTests.java

```diff
@@ -11,50 +11,163 @@
-	private static final String ECLIPSE_INI_FILE_NAME = System.getProperty(ECLIPSE_INI_PATH_KEY,
-			System.getenv(ECLIPSE_INI_PATH_KEY) == null ? "eclipse.ini" : System.getenv(ECLIPSE_INI_PATH_KEY));
-	// @formatter:off
-	private static final String DEFAULT_ECLIPSE_INI_CONTENT = "-startup\n"
-															+ "../test.launcher.jar\n"
-															+ "--launcher.library\n"
-															+ "plugins/org.eclipse.equinox.launcher\n"
-															+ "-vmargs\n"
-															+ "-Xms40m\n"
-															+ "";
-	// @formatter:on
-	public static final Integer EXIT_OK = Integer.valueOf(0);
-	public static final Integer EXIT_RESTART = Integer.valueOf(23);
-	public static final Integer EXIT_RELAUNCH = Integer.valueOf(24);
+
+	private static final String OS_NAME = System.getProperty("os.name").toLowerCase();
+	private static final String ECLIPSE_EXE_NAME = (OS_NAME.contains("win") ? "eclipsec.exe" : "eclipse");
+	private static final String ECLIPSE_INI_FILE_NAME = "eclipse.ini";
+
+	private static final String DEFAULT_ECLIPSE_INI_CONTENT = """
+			-startup
+			test.launcher.jar
+			--launcher.library
+			plugins/org.eclipse.equinox.launcher
+			-vmargs
+			-Xms40m
+			""";
+	public static final Integer EXIT_OK = 0;
+	public static final Integer EXIT_RESTART = 23;
+	public static final Integer EXIT_RELAUNCH = 24;
+
+	@TempDir
+	static Path tempDir;
+	private static Path eclipseInstallationMockLocation;
+
+	@BeforeAll
+	static void prepareLauncherSetup() throws Exception {
+		eclipseInstallationMockLocation = Files.createDirectories(tempDir.resolve("eclipse"));
+
+		Path codeSource = Path.of(LauncherTests.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+		Path equinoxRepo;
+		Path classesDirectory;
+		if (Files.isDirectory(codeSource) && codeSource.endsWith(Path.of("org.eclipse.equinox.launcher.tests"))) {
+			// Test executed from within the IDE
+			equinoxRepo = codeSource.getParent().getParent();
+			classesDirectory = Path.of("bin");
+		} else if (Files.isDirectory(codeSource)
+				&& codeSource.endsWith(Path.of("org.eclipse.equinox.launcher.tests/target/classes"))) {
+			// Test executed by Maven+Tycho
+			equinoxRepo = codeSource.getParent().getParent().getParent().getParent();
+			classesDirectory = Path.of("target", "classes");
+		} else {
+			throw new IllegalStateException("Unkown state");
+		}
+
+		Path binariesRoot = findBinariesRoot(equinoxRepo);
+		if (!Files.isDirectory(binariesRoot)) {
+			throw new IllegalArgumentException("Supplied equinox binaries location does not exist: " + binariesRoot);
+		}
+		String os = System.getProperty("osgi.os");
+		String ws = System.getProperty("osgi.ws");
+		String arch = System.getProperty("osgi.arch");
+
+		// Assemble a Eclipse installation mock in the test-temporary directory
+
+		Path exePath = Path.of("org.eclipse.equinox.executable", "bin", ws, os, arch);
+		if (os.contains("mac")) {
+			exePath = exePath.resolve("Eclipse.app/Contents/MacOS/eclipse");
```

Comment:
can we move `"Eclipse.app/Contents/MacOS/eclipse"` as value of `ECLIPSE_EXE_NAME` in case of mac?

Reply:
The problem is that for the test-setup it looks like it didn't work if the exe was moved to that sub-directory, but maybe I have to check that again.
I have no Mac and can therefore only test it in the CI.